### ISSUE TITLE
Ways to extract docker digest in Pack cli build.

### ISSUE THREAD

**Issue by anoopvlcy** [buildpacks/pack#1565]
### Description
Docker Digest is immutable compare to docker tags. Hence the best practice in CICD is to use Docker Digest instead of docker tags. 
Currently Pack CLI Build command doesn't provide any _programmatically_ readable ways to extract docker digest.

Pack Build with "-q" provide provide docker digest, but the command output can container other information as well (for example any warning or errors etc). So we need a text parsing to extract docker digest (its a ugly way doing things.) Also in CICD it is preferred to have detailed logs rather 'quite' mode with less information in it.

The build tools such as [Kanikno](https://github.com/GoogleContainerTools/kaniko/blob/main/README.md), Docker Inspect provides a way to either retrieve digest(even CNB "Creator" have an option to generate report).

### Proposed solution
Provide a way to generate report like cnb 'creator'

### Describe alternatives you've considered
use pack cli with '-q' and text parsing

**Comment by natalieparellano** [comment]
Hi @anoopvlcy - I believe the eventual completion of https://github.com/buildpacks/pack/issues/1221 would solve this issue. I myself have been wanting this for some time. If you (or anyone else) would be interested in working on it, I'd be happy to guide you.

**Comment by h20220025** [comment]
@anoopvlcy can i work on this ??

**Comment by anoopvlcy** [comment]
> @anoopvlcy can i work on this ??

hi @h20220025 
Yes. Please. No issues from myside. :)
cc @natalieparellano

**Comment by jjbustamante** [comment]
As @natalieparellano mentioned, I think this issue is already solved after merging #1221

I agree that the user will have the `--report-output-dir` flag to define where you want to save the `report.toml` file and inside you will get the Digest you are looking for. 

As an example (because it is not released yet), I built `pack` from the main branch and run the following:

```bash
 pack git:(main) ✗ ./out/pack build sample-app --path ../samples/apps/java-maven --builder cnbs/sample-builder:bionic --report-output-dir .
```

After the image is built, I got the `report.toml` file in my directory:

```toml
[build]

[image]
  tags = ["sample-app"]
  image-id = "9f8c2efbb5e479bf0048fca5dbe9fc0204341b50386b6bebf4688115fe2f1cda"
```

Users can read the report file and extract the `image-id`
